CMakeLists.txt

```diff
@@ -179,7 +181,7 @@ IF(UNIX)
     IF(APPLE)
       TARGET_LINK_LIBRARIES(${PROJECT_NAME} ${IPOPT_LIBRARY_DIRS}/libipopt.dylib)
     ELSE()
-      TARGET_LINK_LIBRARIES(${PROJECT_NAME} ipopt)
+      TARGET_LINK_LIBRARIES(${PROJECT_NAME} ${IPOPT_LIBRARY_DIRS}/libipopt.so)
```

Comment:
Normally, we should be platform independent for the linkage. This is not a good fix.

Reply:
I thought it was a bit iffy. Maybe we should be using `find_library()` to get the full path to the lib directly?

Reply:
No, you need to add the LIBRARY_DIRS to the target via target_link_directories and just link to `ipopt`.